### PR TITLE
[HIBIKE] Servo Control

### PR DESCRIPTION
**Pull request by JHGZhou**

### CODE REVIEW

File: hibike/devices/ServoControl/servo_control.cpp [hunk 1]
```diff
@@ -33,73 +28,26 @@ void loop() {
...
-}
-
-// you must implement this function. It is called when the deviCe receives a DeviceStatus packet.
-// the return value is the value field of the DeviceRespond packet hibike will respond with
-uint32_t device_status(uint8_t param) {
-  if (param < NUM_PINS) {
-    return servos[param].read();
+uint32_t device_write(uint8_t param, uint8_t* data, size_t len) {
+  float value = ((float *)data)[0];
+  if (value < -1 || value > 1) {
```
Comment (dHutchings) [inline hibike/devices/ServoControl/servo_control.cpp]: Currently, if the value is not in an acceptable range, it's thrown away.  Is that the desired implimentation?

File: hibike/devices/ServoControl/servo_control.cpp [hunk 2]
```diff
@@ -33,73 +28,26 @@ void loop() {
...
-      toggle3 = value;
-      return toggle_servo(0, toggle3);
-      break;
-
-    default:
-      return ~((uint32_t) 0);
+uint32_t device_write(uint8_t param, uint8_t* data, size_t len) {
+  float value = ((float *)data)[0];
+  if (value < -1 || value > 1) {
+    return sizeof(float);
```
Comment (dHutchings) [inline hibike/devices/ServoControl/servo_control.cpp]: Current implementation will have the device throw away any instruction that's not in the acceptable range.  Is this behavior intended?

File: hibike/devices/ServoControl/servo_control.cpp [hunk 3]
```diff
@@ -33,73 +28,26 @@ void loop() {
...
 }
 
 // You must implement this function.
 // It is called when the BBB sends a message to the Smart Device tellinng the Smart Device to disable itself.
 // Consult README.md, section 6, to see what exact functionality is expected out of disable.
 void device_disable() {
-
-}
+  disableAll();
+}
```
Comment (dHutchings) [inline hibike/devices/ServoControl/servo_control.cpp:67]: newline at EOF

File: hibike/devices/ServoControl/servo_control.h [hunk 4]
```diff
@@ -4,37 +4,24 @@
...
-#define SERVO_2 10
-#define SERVO_3 11
+#define NUM_PINS 2
+#define SERVO_0 5
+#define SERVO_1 6
 
 #define LED_PIN 13
+
+#define SERVO_RANGE 1000
+#define SERVO_CENTER 1500
```
Comment (dHutchings) [inline hibike/devices/ServoControl/servo_control.h:14]: I approve of defining these constants

File: hibike/devices/ServoControl/servo_control.h [hunk 5]
```diff
@@ -4,37 +4,24 @@
...
-#define NUM_PINS 4
-#define SERVO_0 6
-#define SERVO_1 9
-#define SERVO_2 10
-#define SERVO_3 11
+#define NUM_PINS 2
+#define SERVO_0 5
+#define SERVO_1 6
 
 #define LED_PIN 13
```
Comment (dHutchings) [inline hibike/devices/ServoControl/servo_control.h:11]: Is this LED pin used for anything?  Ir not, remove it plz.

File: hibike/devices/ServoControl/servo_control.cpp [hunk 6]
```diff
@@ -33,73 +28,26 @@ void loop() {
 
 // you must implement this function. It is called when the device receives a DeviceUpdate packet.
 // the return value is the value field of the DeviceRespond packet hibike will respond with
```
Comment (kevinji) [inline hibike/devices/ServoControl/servo_control.cpp]: This comment no longer seems accurate; can you update it?

**Review by dHutchings — APPROVED** [review]
LGTM.  Good job!

**Pull request merged**